commands/import_e2e_test.go

```diff
@@ -0,0 +1,309 @@
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Len(printer.GetErrorLines(), 0)
+		s.Require().Equal(us.Id+"_"+importName, printer.GetLines()[0].(*model.Job).Data["import_file"])
+	})
+}
+
+func (s *MmctlE2ETestSuite) TestImportListAvailableCmdF() {
+	s.SetupTestHelper().InitBasic()
+	serverPath := os.Getenv("MM_SERVER_PATH")
+	importName := "import_test.zip"
+	importFilePath := filepath.Join(serverPath, "tests", importName)
+
+	s.Run("no permissions", func() {
+		printer.Clean()
+
+		err := importListAvailableCmdF(s.th.Client, &cobra.Command{}, nil)
+		s.Require().NotNil(err)
+		s.Require().Equal("failed to list imports: : You do not have the appropriate permissions., ", err.Error())
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.RunForSystemAdminAndLocal("no imports", func(c client.Client) {
+		printer.Clean()
+
+		err := importListAvailableCmdF(c, &cobra.Command{}, nil)
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Len(printer.GetErrorLines(), 0)
+		s.Equal("No import files found", printer.GetLines()[0])
+	})
+
+	s.RunForSystemAdminAndLocal("some imports", func(c client.Client) {
+		cmd := &cobra.Command{}
+		if c == s.th.LocalClient {
+			cmd.Flags().Bool("local", true, "")
+		}
+
+		numImports := 3
+		for i := 0; i < numImports; i++ {
+			err := importUploadCmdF(c, cmd, []string{importFilePath})
+			s.Require().Nil(err)
+		}
+		printer.Clean()
+
+		imports, appErr := s.th.App.ListImports()
+		s.Require().Nil(appErr)
+
+		err := importListAvailableCmdF(c, cmd, nil)
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), len(imports))
+		s.Require().Len(printer.GetErrorLines(), 0)
+		for i, name := range printer.GetLines() {
+			s.Require().Equal(imports[i], name.(string))
+		}
+	})
+}
+
+func (s *MmctlE2ETestSuite) TestImportListIncompleteCmdF() {
+	s.SetupTestHelper().InitBasic()
+
+	s.RunForAllClients("no incomplete import uploads", func(c client.Client) {
+		printer.Clean()
+
+		err := importListIncompleteCmdF(c, &cobra.Command{}, nil)
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Len(printer.GetErrorLines(), 0)
+		s.Equal("No incomplete import uploads found", printer.GetLines()[0])
+	})
+
+	s.RunForSystemAdminAndLocal("some incomplete import uploads", func(c client.Client) {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+		userID := "nouser"
+		if c == s.th.SystemAdminClient {
+			user, resp := s.th.SystemAdminClient.GetMe("")
+			s.Require().Nil(resp.Error)
+			userID = user.Id
+		} else {
+			cmd.Flags().Bool("local", true, "")
+		}
+
+		us1, appErr := s.th.App.CreateUploadSession(&model.UploadSession{
+			Id:       model.NewId(),
+			UserId:   userID,
+			Type:     model.UploadTypeImport,
+			Filename: "import1.zip",
+			FileSize: 1024 * 1024,
+		})
+		s.Require().Nil(appErr)
+		us1.Path = ""
+
+		time.Sleep(time.Millisecond)
```

Comment:
Not that I'm opposed to it, but why do we need the sleep in between uploads?

Reply:
That is to avoid weird bugs (which cause flakiness) where two upload sessions get created at the exact same time and the resulting order changes.